readthedocs/gold/forms.py

```diff
@@ -90,7 +91,13 @@ def __init__(self, *args, **kwargs):
 
     def clean(self):
         cleaned_data = super(GoldProjectForm, self).clean()
+        project_slug = cleaned_data.get('project', "")
         if self.projects.count() < self.user.num_supported_projects:
-            return cleaned_data
-
-        self.add_error(None, 'You already have the max number of supported projects.')
+            # Checking if the project with the entered slug
+            # is present in the database or not
+            if Project.objects.filter(slug=project_slug).exists():
```

Comment:
Maybe this fits better on a `clean_project` method?

Reply:
yeahh... that will be better.
I have implemented it.